FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,52 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url, completion: { [weak self] result in
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, httpResponse)):
+				if httpResponse.statusCode == 200,
+				   let root = try? JSONDecoder().decode(Root.self, from: data) {
+					completion(.success(root.items.map { $0.feedImage }))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		})
+	}
+
+	private struct Root: Decodable {
+		let items: [APIFeedImage]
+	}
+
+	private struct APIFeedImage: Hashable, Decodable {
```

Comment:
There's no need to conform to `Hashable`, so this conformance can be removed.
```suggestion
	private struct APIFeedImage: Decodable {
```